src/RolloutScorer/RolloutScorer/RolloutScorer.cs

```diff
@@ -49,85 +50,93 @@ public async Task InitAsync()
             // Convert the rollout start time and end time to the strings the AzDO API recognizes and fetch builds
             string rolloutStartTimeUriString = RolloutStartDate.ToUniversalTime().ToString("yyyy-MM-ddTHH:mm:ssZ");
             string rolloutEndTimeUriString = RolloutEndDate.ToUniversalTime().ToString("yyyy-MM-ddTHH:mm:ssZ");
-            JObject responseContent = await GetAzdoApiResponseAsync($"https://dev.azure.com/{RepoConfig.AzdoInstance}/" +
-                $"{AzdoConfig.Project}/_apis/build/builds?definitions={RepoConfig.DefinitionId}&branchName={Branch}" +
-                $"&minTime={rolloutStartTimeUriString}&maxTime={rolloutEndTimeUriString}&api-version=5.1");
 
-            // No builds is a valid case (e.g. a failed rollout) and so the rest of the code can handle this
-            // It still is potentially unexpected, so we're going to warn the user here
-            if (responseContent.Value<int>("count") == 0)
+            foreach (string buildDefinitionId in RepoConfig.BuildDefinitionIds)
             {
-                Utilities.WriteWarning($"No builds were found for repo '{RepoConfig.Repo}' " +
-                        $"(Build ID: '{RepoConfig.DefinitionId}') during the specified dates ({RolloutStartDate} to {RolloutEndDate})", Log);
-            }
+                JObject responseContent = await GetAzdoApiResponseAsync($"https://dev.azure.com/{RepoConfig.AzdoInstance}/" +
+                $"{AzdoConfig.Project}/_apis/build/builds?definitions={buildDefinitionId}&branchName={Branch}" +
+                $"&minTime={rolloutStartTimeUriString}&maxTime={rolloutEndTimeUriString}&api-version=5.1");
 
-            JArray builds = responseContent.Value<JArray>("value");
-            foreach (JToken build in builds)
-            {
-                BuildBreakdowns.Add(new ScorecardBuildBreakdown(build.ToObject<BuildSummary>()));
+                // No builds is a valid case (e.g. a failed rollout) and so the rest of the code can handle this
+                // It still is potentially unexpected, so we're going to warn the user here
+                if (responseContent.Value<int>("count") == 0)
+                {
+                    Utilities.WriteWarning($"No builds were found for repo '{RepoConfig.Repo}' " +
+                            $"(Build ID: '{buildDefinitionId}') during the specified dates ({RolloutStartDate} to {RolloutEndDate})", Log);
+                }
+
+                JArray builds = responseContent.Value<JArray>("value");
+                foreach (JToken build in builds)
+                {
+                    BuildBreakdowns.Add(new ScorecardBuildBreakdown(build.ToObject<BuildSummary>()));
+                }
             }
+            BuildBreakdowns.Sort((x, y) => x.BuildSummary.FinishTime.CompareTo(y.BuildSummary.FinishTime));
```

Comment:
I'd probably do `BuildBreakdowns = BuildBreakdowns.OrderBy(x => x.BuildSummary.FinishTime)` myself... orderBy is a lot easier to wrap my head around than the sort with the manual CompareTo.  But that's just a preference thing.

Reply:
Does it need to be sorted?  I don't see where the sorting actually affects anything, but maybe I'm missing something subtle.

---
In reply to: [370361664](https://github.com/dotnet/arcade-services/pull/903#discussion_r370361664) [](ancestors = 370361664)

Reply:
OrderBy is a lot easier, you're right -- I think I will swap to that. Sorting affects failure calculation because we specifically look at the *last* build to have a deployment and see if that's green; AzDO sorts from most recent to least recent and we want the opposite of that. Also, I don't want to rely on AzDO for sorting, so this is a safety measure as well.